### PR TITLE
docs: Added FAQs to assist with common issues with next.js instrumentation

### LINKED ISSUE #181: Update README and include FAQ around common problems
## Is your feature request related to a problem? Please describe.
We did an analysis of issues logged in this project.  Most start with an ask that quicky diverges into common pain points.  We need to provide clear context about what this supports and what it does not support.

* Update README to include reference to FAQ
* Create FAQs around the following:
   * document that deploying to cloud providers like Vercel, AWS Amplify, Netlify, Azure Static sites is not supported.  Reference the file to demonstrate how to setup otel and send data via OTLP to New Relic
   * how to inject browser agent
   * how to get 3rd party instrumentation loaded via webpack externals
* Update `Client-side Instrumentation` section of README and link to the examples apps that demonstrate how to inject browser agent and handling errors in legacy and app router Next.js apps(see https://github.com/newrelic/newrelic-node-examples/issues/202 and https://github.com/newrelic/newrelic-node-examples/issues/203)
* Update links to the sample application(s) that point to the examples in `newrelic-node-examples`

### PR DESCRIPTION
**Pull request by bizob2828**
## Description

Added FAQs for common issues. Links to app router examples will follow once the example app is done(see https://github.com/newrelic/newrelic-node-examples/issues/203)

## Related Issues

Closes #181

### CODE REVIEW

File: docs/faqs/cloud-providers.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Deploy Next.js to Cloud Provider
+
+Q: Can `@newrelic/next` work when deploying to [Vercel](https://vercel.com/frameworks/nextjs), [AWS Amplify](https://aws.amazon.com/amplify/), [Netlify](https://www.netlify.com/with/nextjs/), [Azure Static Sites](https://azure.microsoft.com/en-us/products/app-service/static), etc?
+
+A: The short answer is no. Most of these cloud providers lack the ability to control run options to load `@newrelic/next`.  Also, most of these cloud providers execute code in a Functions as a Service(FaaS) environment.  Our agent requires different setup and then additional processes to load the telemetry.  Our recommendation is to rely on OpenTelemetry and load the telemetry via our OTLP endpoint.
```
Comment (jsumners-nr) [inline docs/faqs/cloud-providers.md:5]:
```suggestion
A: The short answer is no. Most of these cloud providers lack the ability to control run options to load `@newrelic/next`.  Also, most of these cloud providers execute code in a Function as a Service(FaaS) environment.  Our agent requires a different setup and then additional processes to load the telemetry.  Our recommendation is to rely on OpenTelemetry and load the telemetry via our OTLP endpoint.
```

File: docs/faqs/error-handling.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+A: The Node.js agent has an API to log errors `newrelic.noticeError`. Next.js has an error page that can be used to add the API call. 
+
+
+## Log errors to Errors Inbox
+
+The error page varies between [Pages Router](https://nextjs.org/docs/pages/building-your-application/routing/custom-error) and [App Router](https://nextjs.org/docs/app/building-your-application/routing/error-handling) Next.js projects.
+
+
+ * [Pages Router](https://github.com/newrelic/newrelic-node-examples/blob/main/nextjs-legacy/pages/_error.jsx) error handling example.
```
Comment (jsumners-nr) [inline docs/faqs/error-handling.md:13]: I think the link should be to a specific commit hash. It's possible for a `blob/main/` link to become invalid.

File: docs/faqs/browser-agent.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Injecting Browser Agent
+
+Q: How can I inject the [New Relic Browser Agent](https://docs.newrelic.com/docs/browser/browser-monitoring/installation/install-browser-monitoring-agent/) into a Next.js project?
+
+A: It depends on if you are using the Pages or App Router for Next.js. 
+
+
+## Inject Browser Agent with Pages Router
+
+This [Pages Router](https://github.com/newrelic/newrelic-node-examples/blob/main/nextjs-legacy/pages/_document.jsx) example app demonstrates how to inject the browser agent.
```
Comment (jsumners-nr) [inline docs/faqs/browser-agent.md:10]: I think the link should be to a specific commit hash. It's possible for a blob/main/ link to become invalid.

File: docs/faqs/instrument-third-party-libraries.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Instrument 3rd Party Libraries within Next.js 
+
+Q: How can I get instrumentation to load for 3rd parties libraries within my Next.js application like mysql, mongodb, pino, winston, etc?  
+
+A: Typically the New Relic Node.js agent auto-instruments all supported [3rd party libraries](https://docs.newrelic.com/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent/#instrument).  However, Next.js bundles your Next.js project and code spilts between server and client side via webpack.  To get auto-instrumentaiton to work, you must externalize all libraries within webpack.  
+
+## Externalize 3rd party libraries in webpack
+
+To external all supported 3rd party libraries, add the following to `next.config.js`:
```
Comment (mrickard) [inline docs/faqs/instrument-third-party-libraries.md:9]:
```suggestion
To externalize all supported 3rd party libraries, add the following to `next.config.js`:
```

File: docs/faqs/README.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# FAQs
+
+Are you having an issue with New Relic Next.js Instrumentation? Take a look at the following FAQS:
+
+ * [Deploy Next.js to Cloud Provider](./cloud-providers.md)
```
Comment (mrickard) [inline docs/faqs/README.md:5]:
```suggestion
 * [Deploying Next.js to Cloud Provider](./cloud-providers.md)
```

File: docs/faqs/README.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+ * [Injecting New Relic Browser Agent](./browser-agent.md)
+ * [Instrument 3rd Party Libraries](./instrument-third-party-libraries.md)
```
Comment (mrickard) [inline docs/faqs/README.md:7]:
```suggestion
 * [Instrumenting 3rd Party Libraries](./instrument-third-party-libraries.md)
```

**Review by jsumners-nr — APPROVED** [review]
Looks good to me.

File: docs/faqs/instrument-third-party-libraries.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# Instrument 3rd Party Libraries within Next.js 
+
+Q: How can I get instrumentation to load for 3rd parties libraries within my Next.js application like mysql, mongodb, pino, winston, etc?  
+
+A: Typically the New Relic Node.js agent auto-instruments all supported [3rd party libraries](https://docs.newrelic.com/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent/#instrument).  However, Next.js bundles your Next.js project and code spilts between server and client side via webpack.  To get auto-instrumentaiton to work, you must externalize all libraries within webpack.
```
Comment (mrickard) [inline docs/faqs/instrument-third-party-libraries.md:5]:
```suggestion
A: Typically the New Relic Node.js agent auto-instruments all supported [3rd party libraries](https://docs.newrelic.com/docs/apm/agents/nodejs-agent/getting-started/compatibility-requirements-nodejs-agent/#instrument).  Next.js, however, bundles your project and code spilts between server and client side via webpack.  To get auto-instrumentation to work, you must externalize all libraries within webpack.  
```

File: docs/faqs/instrument-third-party-libraries.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# Instrument 3rd Party Libraries within Next.js 
+
+Q: How can I get instrumentation to load for 3rd parties libraries within my Next.js application like mysql, mongodb, pino, winston, etc?
```
Comment (mrickard) [inline docs/faqs/instrument-third-party-libraries.md:3]:
```suggestion
Q: How can I get instrumentation to load for 3rd party libraries within my Next.js application like mysql, mongodb, pino, winston, etc?  
```

**Review by mrickard — APPROVED** [review]

**Pull request merged**